selfdrive/selfdrived/events.py

```diff
@@ -65,9 +67,20 @@ def add(self, event_name: int, static: bool=False) -> None:
       bisect.insort(self.static_events, event_name)
     bisect.insort(self.events, event_name)
 
+    self.event_counters[event_name] += 1
+
   def clear(self) -> None:
-    self.event_counters = {k: (v + 1 if k in self.events else 0) for k, v in self.event_counters.items()}
+     # Get events no longer active
```

Comment:
This looks immediately more confusing to read, anything else we can try?

Reply:
A more common and readable approach is to loop through event_counters and check if event_name is not in the events list. The code snippet would look like this:
```
for event_name in self.event_counters:
    if event_name not in self.events:
        self.event_counters[event_name] = 0
```
However, this approach is significantly less efficient than the version in this PR and might not offer much improvement over the master version. I'll consider if there's a better solution.

Reply:
I can't think of a better solution for now. I've improved the comments to make the code easier to understand. The logic is straightforward: maintain a list of previously active events, compare it with the current active events, and remove those that are no longer active.